src/components/Calculator.js

```diff
@@ -0,0 +1,91 @@
+import React, { useState } from 'react';
+import './Calculator.css';
+import Button from './Button';
+
+const Calculator = () => {
+  const [displayValue, setDisplayValue] = useState('0');
+  const [prevValue, setPrevValue] = useState(null);
+  const [operator, setOperator] = useState(null);
+
+  const calculate = (num1, num2, op) => {
+    const value1 = parseFloat(num1);
+    const value2 = parseFloat(num2);
+
+    switch (op) {
+      case '+':
+        return value1 + value2;
+      case '-':
+        return value1 - value2;
+      case '*':
+        return value1 * value2;
+      case '/':
+        return value1 / value2;
+      default:
+        return value2;
+    }
+  };
+
+  const handleButtonClick = (value) => {
+    if (value === 'C' || value === 'AC') {
+      // Clear the display and reset state
+      setDisplayValue('0');
+      setPrevValue(null);
+      setOperator(null);
+    } else if (['+', '-', '*', '/'].includes(value)) {
+      // Update prevValue and operator when an operator is pressed
+      setPrevValue(displayValue);
+      setOperator(value);
+      setDisplayValue('');
+    } else if (value === '=') {
+      // Perform calculation when '=' is pressed
+      if (prevValue && operator) {
+        const result = calculate(prevValue, displayValue, operator);
+        setDisplayValue(result.toString());
+        setPrevValue(null);
+        setOperator(null);
+      }
+    } else if (value === '.' && displayValue.includes('.')) {
+      // Avoid adding multiple decimal points
+
+    } else {
+      // Update display value
+      setDisplayValue((prevDisplay) => (prevDisplay === '0' || operator === prevDisplay ? value : prevDisplay + value));
+    }
+  };
+
+  // const handleButtonKeyPress = (event, value) => {
+  //   if (event.key === 'Enter') {
+  //     handleButtonClick(value);
+  //   }
+  // };
+
```

Comment:
- [ ] Kindly remove these commented lines of code. It was not present during the previous review. Committing commented code is not good practice as it can make your code hard to read and maintain.

Reply:
well resolved! ✔️